fenix/app/src/main/java/org/mozilla/fenix/shopping/ShoppingExperienceFeature.kt

```diff
@@ -1,19 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-package org.mozilla.fenix.shopping
-
-import org.mozilla.fenix.nimbus.FxNimbus
-
-/**
- * An abstraction for shopping experience feature flag.
- */
-class ShoppingExperienceFeature {
-
-    /**
-     * Returns true if the shopping experience feature is enabled.
-     */
-    val isEnabled
-        get() = FxNimbus.features.shoppingExperience.value().enabled
```

Comment:
Could we just use `settings.enableShoppingExperience` here so when `Settings` is removed we only have to change this reference and none of the feature and test code has to change since it's abstracted?

Reply:
We certainly could, but `Settings` would still need to be passed into here and the test, right? Either it being a property of `ShoppingExperienceFeature` or a parameter of `isEnabled`?  

Unless `ShoppingExperienceFeature` was made into an interface instead 🤔

Reply:
Yeah, settings would be injected here, having an interface would be great 👍 

Tangential but sharing as it's related, some tests are intermittent because of [mockito](https://firefoxci.taskcluster-artifacts.net/dNyDrlhfTzKsp3S4-b7HRQ/0/public/reports/test/testFenixDebugUnitTest/classes/org.mozilla.fenix.shopping.ReviewQualityCheckFeatureTest.html#WHEN%20feature%20is%20enabled%20and%20selected%20tab%20is%20switched%20to%20not%20a%20product%20page%20THEN%20callback%20returns%20false) usage. @gabrielluong and I were thinking of changing it back to a function to be sure mockito doesn't fail here mocking a val. Not 100% sure if mocking a val is an issue here. Or we can use a fake for tests so we don't need to mock.